src/libs/tools/gamepad.c

```diff
@@ -304,7 +318,7 @@ void gamepad_close_devices(dt_lib_module_t *self)
   g_slist_free_full(self->data, (void (*)(void *))gamepad_device_free);
   self->data = NULL;
 
-  SDL_Quit();
+  // SDL_Quit();
```

Comment:
Is that some debug left over?

Reply:
> I did push an extra change here; after adjusting midi parameters or adding devices, the input system can be restarted with ctrl+shift+alt+I. This first closed down SDL2 via SDL_Quit, on the assumption it could then restarted again with SDL_Init but that doesn't work. So I've commented out that one line and now it works fine for me (on Debian); adding controllers and restarting makes them available. The order is somewhat random, but since I highly doubt many people are using dt as a multi-player game, I'm not too bothered...